### DISCUSSION TITLE
Using convolution as crossover.

### DISCUSSION THREAD

**Discussion by Aragorus** [mikebrady/shairport-sync#1605] [Ideas]
If one is to configure shairport-sync in mono mode:
_playback_mode = "mono";_
Does the summation happen before or after the convolution?

If the summing happens before the convolution, could one then use a stereo _"impulse.wav"_ file,
such as for example the left channel is a high-pass filter and right channel is a low-pass filter?

Regards
A

**Comment by Aragorus** [comment]
So if anyone else is curious - it works!
I think it's a very budget friendly way to make a wireless active crossover with a very long FIR filter on each channel.

**Reply by mikebrady** [reply]
Thanks for this, and apologies for not replying to your original question -- I was away from my systems. Can you give any more details, e.g. about the filter?

**Reply by Aragorus** [reply]
Hi Mike,

If one is to use shairport-sync as a mono receiver, it's easy to use REW, or RePhase to generate a couple of filters and save them as .wav files instead of FIR coefficients.
The two mono .wav files can then be joined in a stereo .wav file. One panned left and one right.
This way each channel will be convolved with the coresponding filter.

I used it for an open baffle speaker that needed both delay, EQ and some phase compensation.
It worked as a charm, since a 1s filter which doesn't strain a raspberry pi 3 contains 44100 taps per channel.
Now I'm sure that for most uses 4410 taps is plenty, so one could do with a 0.1s filters instead.

Hope that is helpful!

**Reply by mikebrady** [reply]
I may be coming back to you soon about this...
